### PR TITLE
enh(api): Add extra parameters to update openid configuration auto import

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

**Fixes** # MON-12598

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -16814,3 +16814,9 @@ msgstr "Destination du journal"
 
 msgid "Error"
 msgstr "Erreur"
+
+msgid "Missing auto import mandatory parameters: %s"
+msgstr "Paramètres obligatoires manquants pour l'importation automatique : %s"
+
+msgid "The contact template: '%s' doesn't exist"
+msgstr "Le modèle de contact : '%s' n'existe pas"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16822]:
```suggestion
msgid "The contact template '%s' doesn't exist"
msgstr "Le modèle de contact '%s' n'existe pas"
```

File: src/Core/Application/Security/ProviderConfiguration/OpenId/UseCase/UpdateOpenIdConfiguration/UpdateOpenIdConfiguration.php [hunk 2]
```diff
@@ -40,14 +42,15 @@ class UpdateOpenIdConfiguration
     /**
      * @param WriteOpenIdConfigurationRepositoryInterface $repository
```
Comment (kduret) [inline src/Core/Application/Security/ProviderConfiguration/OpenId/UseCase/UpdateOpenIdConfiguration/UpdateOpenIdConfiguration.php:43]: update phpdoc

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Exceptions/OpenIdConfigurationException.php [hunk 3]
```diff
@@ -26,20 +26,50 @@
...
+    /**
+     * Exception thrown when contact template link to configuration doesn't exist
+     *
+     * @param string $contactTemplateName
+     * @return self
+     */
+    public static function contactTemplateDoesntExist(string $contactTemplateName): self
+    {
+        return new self(_(sprintf(
+            "The contact template: '%s' doesn't exist",
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Exceptions/OpenIdConfigurationException.php:71]:
```suggestion
            "The contact template '%s' doesn't exist",
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfigurationFactory.php [hunk 4]
```diff
@@ -40,6 +41,31 @@ public static function createFromRequest(UpdateOpenIdConfigurationRequest $reque
...
+            }
+            if (empty($request->userAliasBindAttribute)) {
+                $missingMandatoryParameters[] = 'alias_bind_attribute';
+            }
+            if (empty($request->userNameBindAttribute)) {
+                $missingMandatoryParameters[] = 'fullname_bind_attribute';
+            }
+            if (! empty($missingMandatoryParameters)) {
+                throw OpenIdConfigurationException::missingAutoImportMandatoryParameters($missingMandatoryParameters);
+            }
```
Comment (jeremyjaouen) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfigurationFactory.php:66]: move this into model constructor

**Review by callapa — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**